cse/lang/NltkStemmer.py

```diff
@@ -19,9 +19,9 @@ def porter2():
         return SnowballStemmer("english")
 
 
-    def processAll(self, tokens):
-        return [self.__stemmerType.stem(token) for token in tokens]
+    def processAll(self, tokenTuples):
+        return [(self.__stemmerType.stem(token), position) for token, position in tokenTuples]
 
 
-    def process(self, token):
-        return self.__stemmerType.stem(token)
\ No newline at end of file
+    def process(self, tokenTuple):
+        return (self.__stemmerType.stem(tokenTuple[0]), tokenTuple[1])
```

Comment:
Isn't `tokenTuple[1]` the positon o the token in `tokenTuple[0]`

Reply:
I don't understand your question. A `tokenTuple` in the preprocessing pipeline consists of two elements: `(token, position)`, `tokenTuple[0]` refers to `token` and `tokenTuple[1]` refers to `position``

To be consistent over all preprocessing steps I decided to let every step expect a token tuple as a parameter. Therefore the return type of the processing result must also be a token tuple. The most steps don't touch the position of the token, therefore it is only passed to the result tuple via `return (<sth>, tokenTuple[1])`

Reply:
Ok my fault. I read `.stem(tokenTuple[0], tokenTuple[1])`